dhis-2/dhis-web-api-test/src/test/java/org/hisp/dhis/webapi/controller/AbstractCrudControllerTest.java

```diff
@@ -781,6 +781,19 @@ void testFieldsFilterWithAttributeValues()
 
     }
 
+    @Test
+    void testFieldFilterWithAttribute()
+    {
+        Attribute attribute = createAttribute( 'A' );
+        attribute.setDataElementAttribute( true );
+        manager.save( attribute );
+
+        JsonList<JsonIdentifiableObject> response = GET(
+            "/attributes?fields=id,name&filter=dataElementAttribute:eq:true" )
+                .content().getList( "attributes", JsonIdentifiableObject.class );
+        assertEquals( attribute.getUid(), response.get( 0 ).getId() );
```

Comment:
I think to be safe we need a test setup with another attribute created that is not of the type we want to filter out.

Reply:
added one more test case now